daemon/redis.c

```diff
@@ -1582,7 +2058,66 @@ int redis_restore(struct callmaster *m, struct redis *r) {
 }
 
 
+static int json_update_crypto_params(JsonBuilder *builder, const char *pref, const str *callid,
+		unsigned int unique_id,
+		const char *key, const struct crypto_params *p)
+{
+	char tmp[2048]; ZERO(tmp);
+
+	if (!p->crypto_suite)
+		return -1;
+
+	sprintf(tmp,"%s-crypto_suite",key);
+	json_builder_set_member_name (builder, tmp);
+	ZERO(tmp);
+	json_builder_add_string_value (builder, p->crypto_suite->name);
```

Comment:
There's a lot of code repetition here and below. Always sprintf, _set_member_name and _add_*_value. It should be possible to remove the code duplication using wrapper functions or perhaps macros. I'm surprised glib doesn't provide more convenient functions. I believe this is why I've always used json-c so far :)